packages/kbn-optimizer/limits.yml

```diff
@@ -125,6 +125,6 @@ pageLoadAssetSize:
   visTypeGauge: 24113
   unifiedSearch: 49195
   data: 454087
-  expressionXY: 26500
+  expressionXY: 47156
```

Comment:
this is quite big change, can we do something to reduce it? I run it locally and the actual plugin is ~35KB but maybe we could move things around to make it even smaller?

Reply:
So I recalled that @alexwizp did something to minimize it for Lens - for the expressions that are only internally used for Lens, he doesn't export them in the [lens/common/index](https://github.com/elastic/kibana/blob/main/x-pack/plugins/lens/common/index.ts#L15) file. We could do the same here for the expressions only used by expressionXY, WDYT?

Reply:
Big plus 1 - if we need to make the bodies of the expression functions async for this then I think it's worth it.

Reply:
Expression functions are imported on the setup of the plugin's frontend. So, in any case, even if I'll not export the expression function from `common`, it will inflate the size of the bundle. The only option I see here is sharing strings.

<img width="1403" alt="Screenshot 2022-04-19 at 16 42 54" src="https://user-images.githubusercontent.com/22456368/164017383-e89988c8-3e21-488b-8a02-04be5ea5fb2a.png">

Reply:
@Kunzetsov this is the structure Marta is referring to:
* The actual function (which is exported via `common` top level module: https://github.com/elastic/kibana/blob/e2c6a039f60ae634eaa809b83fb707833065f47f/x-pack/plugins/lens/common/expressions/counter_rate/index.ts#L94 ) doesn't have an inline implementation. It only loads it on use
* The function body is declared separately: https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/x-pack/plugins/lens/common/expressions/counter_rate/counter_rate_fn.ts#L22

Reply:
@flash1293, ok, first, let me reduce the amount of strings and arguments, which are duplicating. Then, I'll come to the body of the expressions. It seems to me, the biggest amount of the bundle content is the args + help.

Reply:
@flash1293, thanks for the explanation.

Reply:
@mbondyra, I've reduced the size of the bundle, shared strings, and added async loading of the expression `fn` body. The only problem is that running the bundle update is adding the limit of `44580`. Changed it to 25000 manually, but that is not a good idea, it seems to me.

Reply:
@mbondyra, the current page load bundle is 28.9KB.